pkg/shares/split_compact_shares.go

```diff
@@ -213,7 +228,7 @@ func (css *CompactShareSplitter) isEmpty() bool {
 // Count returns the number of shares that would be made if `Export` was invoked
 // on this compact share splitter.
 func (css *CompactShareSplitter) Count() (shareCount int) {
-	if !css.isEmptyPendingShare() {
+	if !css.isEmptyPendingShare() && !css.done {
```

Comment:
I think we can remove the `&& !css.done` here if we wiped the pending share in `.Export()` and added this check

```go
// isEmptyPendingShare returns true if the pending share is empty, false otherwise.
func (css *CompactShareSplitter) isEmptyPendingShare() bool {
	if css.pendingShare == nil {
		return true
	}
...
```